hivelink-core/src/main/java/org/apache/iceberg/hivelink/core/schema/MergeHiveSchemaWithAvro.java

```diff
@@ -113,6 +114,11 @@ public Schema list(ListTypeInfo list, Schema partner, Schema elementResult) {
     // if there was no matching Avro list, or if matching Avro list was an option, return an optional list
     boolean shouldResultBeOptional = partner == null || AvroSchemaUtil.isOptionSchema(partner);
     Schema result = Schema.createArray(elementResult);
+    if (partner != null) {
```

Comment:
Minor: can these code be abstracted into a function, so they are not duplicated in two places.

Reply:
good call! I'll update the code.